evalml/pipelines/components/transformers/preprocessing/time_series_featurizer.py

```diff
@@ -129,7 +129,16 @@ def fit(self, X, y=None):
 
         # For the multiseries case, where we only want the start delay lag for the baseline
         if isinstance(y, pd.DataFrame):
-            self.statistically_significant_lags = [self.start_delay]
+            self.statistically_significant_lags = {}
+            for column in y.columns:
+                self.statistically_significant_lags[
+                    column
+                ] = self._find_significant_lags(
+                    y[column],
+                    conf_level=self.conf_level,
+                    start_delay=self.start_delay,
+                    max_delay=self.max_delay,
+                )
```

Comment:
Good point, I just consolidated it.

Reply:
We handle the y being null in `self._find_significant_lags` since we calculate all lags in that function (and just set significant lags to be all_lags if y is None). Should I put it off into it's own separate branch, even though the code would be identical to the case where y is a series? e.g. 

```
        # For the multiseries case, each series ID has individualized lag values
        if isinstance(y, pd.DataFrame):
            self.statistically_significant_lags = {}
            for column in y.columns:
                self.statistically_significant_lags[
                    column
                ] = self._find_significant_lags(
                    y[column],
                    conf_level=self.conf_level,
                    start_delay=self.start_delay,
                    max_delay=self.max_delay,
                )
        elif y is None:
            self.statistically_significant_lags = self._find_significant_lags(
                y,
                conf_level=self.conf_level,
                start_delay=self.start_delay,
                max_delay=self.max_delay,
            )
        else:
            self.statistically_significant_lags = self._find_significant_lags(
                y,
                conf_level=self.conf_level,
                start_delay=self.start_delay,
                max_delay=self.max_delay,
            )
```

Reply:
Ok, sorry for drilling into this so much, but I think I understand now. My new potentially hot take proposal is something like:
```
if y is None:
    self.statistically_significant_lags = np.arange(self.start_delay, self.start_delay +self. max_delay + 1)
else:
    if isinstance(y, pd.Series): 
        y = y.to_frame()
    for column in y.columns:
        self.statistically_significant_lags = ...
```
And then we can remove the handling of y being None from the static function. My argument for doing this is that calling all lags the statistically significant lags is a misnomer, since we didn't actually check statistical significance. This is me getting *very* into the weeds though, so I very much understand if you would rather keep things closer to the way they are 😅 

Regardless, even with your new proposal, we'd still be able to combine the two non y=None cases by casting the series to a dataframe

Reply:
Your example makes sense to me, I don't see our behavior for `y is None` changing anytime soon so I'm comfortable with pulling that out and changing the function.